### DISCUSSION TITLE
retagger not working

### DISCUSSION THREAD

**Discussion by phil-flip** [jonaswinkler/paperless-ng#673] [Support]
Heyo,
so I don't know if this is a bug or anything, but when I run `document_retagger -T -t` after I created some tags and document types. It processes all the Documents, but they are not getting sorted.
I also ran `document_create_classifier` to update the algorithm, but sadly that didn't help either.
I simply have a tag matching "Rechnung" without case sensibility and an auto document type with 6 sample docs for the AI to learn. (they are almost the same).
I updated paperless 4 hours ago or so to the newest docker repository.
Does someone have any advice I should try to get this working?
If any additional info is needed I will gladly provide it.

**Comment by jonaswinkler** [comment]
Do these documents already have tags and types assigned? Maybe this section from the documenation is explaining what's wrong here:

> Finally, -f specifies that you wish to overwrite already assigned correspondents, types and/or tags. The default behavior is to not assign correspondents and types to documents that have this data already assigned. -f works differently for tags: By default, only additional tags get added to documents, no tags will be removed. With -f, tags that don’t match a document anymore get removed as well.

**Reply by phil-flip** [reply]
I just removed all tags, correspondents and document type of the test document and gave it another spin (with the classifier running first). Sadly it didn't work regardless.
Is it a requirement tho? To get the Document processed?

I ran it again with the -f argument. Well as the last sentience says... If it can't be assigned it's going to remove the existing tags.
(No worries tho. I'm still testing around before using it in production)
But it did not assign it to any document type/tag.
The Inbox tag still holds all the documents tho..

**Reply by phil-flip** [reply]
I also uploaded a test document to see, if it actually works at all... But it doesn't (unless this takes some time to kick in).
So this might just be a configuration thing...

**Reply by jonaswinkler** [reply]
Alright, let's see.

* The retagger is one of those things that's still not 100% covered by unit tests, but it hasn't changed in quite a long time and the test cases that *do* exist verify that it's generally able to get the job done. So maybe you just ran into some edge case that I need to cover.
* I've created a single tag over here, set the matching to "Any", set the match to a word that occurs many times in the documents. `document_retagger -T` works.



> Is it a requirement tho? To get the Document processed?

No, not a requirement. More like a safety guard so that the document_retagger does not remove already assigned metadata by default. That's why `-f` exists.



> The Inbox tag still holds all the documents tho..

As per some request in the past #125, the retagger won't touch inbox tags.


> I also uploaded a test document to see, if it actually works at all... But it doesn't (unless this takes some time to kick in).
> So this might just be a configuration thing...

No, they should appear immediately. Combining everything you said really sounds like that your documents actually don't match your tags / types.

One thing that comes to mind, and that's a quirk from paperless (not NG) that I still have not looked at:

* All these matching algorithms work on whole words only (except regex, which just does regex matching, and auto, which is special). So a match `Rechnung` with matching algorithm `Any` will not trigger on documents that contain `Rechnungsadresse` and not `Rechnung` by itself. Maybe check that.

On a side note, the auto matching algorithm also needs a fair bit of negative examples (documents that do *not* have the tag/type/correspondent marked as auto) to work properly. A couple hundret documents is where it usually gets going really well.

**Reply by phil-flip** [reply]
My test document was just a pdf with the Word written 7 times on it, so it should have picked it up. But it didn't.. 

The algorithm has currently a 140+ Documents to play with.

**Reply by jonaswinkler** [reply]
Please upload that test document, and also a screenshot of the edit tag dialog of the tag that you expect to match on this document.

**Reply by phil-flip** [reply]
Yea... I figured it out... It was a configuration mistake just as I thought.
I feel kinda stupid for not reading that it is not "Case sensitive" but "Case insensitive". But on the other hand. I don't quite get it why it didn't recognize it, because some do have it written like that (like my test Document)
Also, the auto Document Type doesn't have any documents at all…

Anyways here is it, if you still wanted to see the config and file.
[test document.pdf](https://github.com/jonaswinkler/paperless-ng/files/6070672/test.document.pdf)
![image](https://user-images.githubusercontent.com/27019397/109691228-852c3380-7b87-11eb-9b3d-26f68fd42781.png)
